### PR TITLE
ISPN-13568 MemcachedFunctionalTest.testTouchWithExpiryUnixTime fix

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-13568

### CODE REVIEW

File: core/src/test/java/org/infinispan/test/TestingUtil.java [hunk 1]
```diff
@@ -1447,9 +1434,9 @@ public static ObjectName getJGroupsChannelObjectName(EmbeddedCacheManager cacheM
       GlobalConfiguration cfg = cacheManager.getCacheManagerConfiguration();
       try {
          return new ObjectName(String.format("%s:type=channel,cluster=%s,manager=%s",
-                                             cfg.jmx().domain(),
-                                             ObjectName.quote(cacheManager.getClusterName()),
-                                             ObjectName.quote(cfg.cacheManagerName())));
+               cfg.jmx().domain(),
+               ObjectName.quote(cacheManager.getClusterName()),
+               ObjectName.quote(cfg.cacheManagerName())));
```
Comment (danberindei) [inline core/src/test/java/org/infinispan/test/TestingUtil.java:1439]: -1, I really like the aligned method parameters (and chained method calls)
Plus, you aligned the parameter descriptions in the javadoc, which seems less useful to me :)

File: server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java [hunk 2]
```diff
@@ -108,14 +109,14 @@ public void testAddBasic(Method m) throws InterruptedException, ExecutionExcepti
...
-      int future = (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() + 1000);
+      int future = (int) TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis() + 2000);
       OperationFuture<Boolean> f = client.set(k(m), future, v(m));
       assertTrue(f.get(timeout, TimeUnit.SECONDS));
       f = client.touch(k(m), future + 1);
       assertTrue(f.get(timeout, TimeUnit.SECONDS));
       sleepThread(1100);
       assertEquals(v(m), client.get(k(m)));
-      sleepThread(1100);
+      sleepThreadUntil(TimeUnit.SECONDS.toMillis(future) + 1000);
```
Comment (danberindei) [inline server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java:119]: -100 to sleep for 3 seconds in a test, we should use ControlledTimeService instead.

File: core/src/test/java/org/infinispan/test/TestingUtil.java (hunk 1 above)
Comment (tristantarrant) [reply core/src/test/java/org/infinispan/test/TestingUtil.java]: I've reverted all changes.

File: server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java (hunk 2 above)
Comment (tristantarrant) [reply server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java]: Converted to use the TimeService

**Comment by tristantarrant** [comment]
Looks good now

**Review by danberindei — COMMENTED** [review]
Looks great, I hadn't considered that maybe the memcached server wasn't even using TimeService.

Getting nit-picky here, but I'd love it if the flush delay was just 1 second and the test actually checked that the keys are still present immediately after the flush operation completes.

File: server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java [hunk 3]
```diff
@@ -469,7 +469,8 @@ private void flushAllDelayed(Method m, int delay, long sleep) throws Interrupted
       OperationFuture<Boolean> f = client.flush(delay);
       assertTrue(f.get(timeout, TimeUnit.SECONDS));
 
-      sleepThread(sleep);
+      sleepThread(sleep); // The underlying ScheduledExecutorService does not use the ControlledTimeService unfortunately
```
[the remark this answers is not in the document]
Comment (tristantarrant) [reply server/memcached/src/test/java/org/infinispan/server/memcached/MemcachedFunctionalTest.java:472]: Nit-picky is good, but this is a deprecated component, and this test has never been a problem.

**Comment by wburns** [comment]
Integrated into main, thanks @tristantarrant !

**Pull request merged**